src/crimson/osd/osd.cc

```diff
@@ -1113,6 +1113,7 @@ seastar::future<> OSD::committed_osd_maps(
   ceph_assert(seastar::this_shard_id() == PRIMARY_CORE);
   INFO("osd.{} ({}, {})", whoami, first, last);
   // advance through the new maps
+  auto old_map = osdmap;
```

Comment:
`old_map` should be also updated within the loop in each iteration, right?
For the case where no only a single map is handled (between `first` to `last+1`).

Reply:
I think that we should identify down OSDs inside the `seastar::do_for_each` loop below. So that we can compare each map to it's previous one.
Do you think that simply comparing the `old_map` to the newest map is enough?

Reply:
> Do you think that only comparing the `old_map` to the newest map will do?

I think we are trying to find live connections the peer of which is down, so maybe only comparing the `old_map` and the newest map is enough. What do you think?

Reply:
> Nay, we should do markdowns for each new map.

Do you mean each step may create new connections, which might not be handled if only looking at the `old_map`?

Reply:
> > Nay, we should do markdowns for each new map.
> 
> Do you mean each step may create new connections, which might not be handled if only looking at the `old_map`?

Um, I meant OSDs may go down and up during the time from old_map to newest_map, which means their nonces are changed, so we need to mark down all those OSDs' connections.